macos/FluentUI/Button.swift

```diff
@@ -0,0 +1,207 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import AppKit
+
+/// Indicates what style our button is drawn as
+@objc public enum ButtonStyle: Int, CaseIterable {
+    case primaryFilled	// Solid fill color
+    case primaryOutline	// Clear fill color, solid outline
+    case borderless		// clear fill color, clear outline
+}
+
+// MARK: - Button
+
+/// A fluent styled button, with hover effects and a corner radius.
+@IBDesignable
+@objc(MSFButton)
+open class Button: NSButton {
+
+	@objc public init(title: String, style: ButtonStyle = .primaryFilled) {
+		super.init(frame: .zero)
+		self.title = title
+		self.style = style
+        initialize()
+    }
+	
+	public override init(frame frameRect: NSRect) {
+        super.init(frame: frameRect)
+        initialize()
+	}
+
+	@available(*, unavailable)
+	required public init?(coder decoder: NSCoder) {
+		preconditionFailure()
+	}
+	
+	open func initialize() {
+		// Do common initialization work
+		isBordered = false
+		wantsLayer = true
```

Comment:
why are we setting wants layer to true when we override the property below?

Reply:
This is (very) confusing, and I recommend reading the [Layer](https://developer.apple.com/documentation/appkit/nsview/1483298-layer), [wantsLayer](https://developer.apple.com/documentation/appkit/nsview/1483695-wantslayer), and [wantsUpdateLayer](https://developer.apple.com/documentation/appkit/nsview/1483461-wantsupdatelayer) documentation for more info, but TL;DR:

Setting `wantsLayer` determines if the view is layer-backed, 
Overriding `wantsUpdateLayer` tells the view whether we use `draw(_:)` or `updateLayer()` to update the contents of the layer.